### PR TITLE
Add telemetry data

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Compose product info string with general info about platform and runtime.

## Motivation and Context
- Provide product usage data upstream.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

cc @rido-min

### CODE REVIEW

File: nanoFramework.Azure.Devices.Client/DeviceClient.cs [hunk 1]
```diff
@@ -160,9 +160,17 @@ public bool Open()
             if (!string.IsNullOrEmpty(ModelId))
             {
                 userName += $"&model-id={HttpUtility.UrlEncode(ModelId)}";
-
             }
 
+            // compose product info
+            StringBuilder productInfo = new StringBuilder();
```
Comment (Ellerbach) [inline nanoFramework.Azure.Devices.Client/DeviceClient.cs:166]: For performance reasons, we've seen with the json class that string concatenations is faster.
So maybe just use $"" all the time?

**Review by Ellerbach — APPROVED** [review]
just a performance comment otherwise all good

**Review by Ellerbach — APPROVED** [review]
Good to go :-)

**Comment by josesimoes** [comment]
> Good to go :-)

Thanks! Let's just wait for @rido-min comments on this before merging.

**Pull request merged**